package/src/components/Channel/hooks/useCreateThreadContext.ts

```diff
@@ -15,38 +23,38 @@ export const useCreateThreadContext = <
   setThreadLoadingMore,
   thread,
   threadHasMore,
+  threadInstance,
   threadLoadingMore,
   threadMessages,
 }: ThreadContextValue<StreamChatGenerics>) => {
-  const threadId = thread?.id;
-  const threadReplyCount = thread?.reply_count;
-  const threadLatestReactions = thread?.latest_reactions;
-  const threadMessagesStr = reduceMessagesToString(threadMessages);
+  const [latestReplies, isLoadingPrev, isLoadingNext] =
+    useStateStore(threadInstance?.state, selector) ?? [];
+
+  // console.log('ISE: LATEST REPLIES TEST: ', threadMessages.length, latestReplies?.length, thread?.activate)
+  // console.log('ISE: MSGS', threadMessages?.length, latestReplies?.length)
 
-  const threadContext: ThreadContextValue<StreamChatGenerics> = useMemo(
-    () => ({
-      allowThreadMessagesInChannel,
-      closeThread,
-      loadMoreThread,
-      openThread,
-      reloadThread,
-      setThreadLoadingMore,
-      thread,
-      threadHasMore,
-      threadLoadingMore,
-      threadMessages,
-    }),
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-    [
-      allowThreadMessagesInChannel,
-      threadHasMore,
-      threadId,
-      threadLoadingMore,
-      threadMessagesStr,
-      threadReplyCount,
-      threadLatestReactions,
```

Comment:
this is uber cool, removal of useMemo 👏 

I hope we do more of this in the SDK in full speed